src/TestInfrastructure/VsObjectModelMocks/LegacyProjectMock.cs

```diff
@@ -0,0 +1,98 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell.Flavor;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests
+{
+    /// <summary>
+    /// Mock for managed projects that use the legacy C#/VB project system
+    /// </summary>
+    public class LegacyProjectMock : ProjectMock, IVsAggregatableProjectCorrected
```

Comment:
I extracted this class from the existing `ProjectMock` class - no other changes.
The legacy mock implements the `IVsAggregatableProjectCorrected` interface that we use to detect which project system is being used.